### PR TITLE
(social-media-login) has been implemented

### PR DESCRIPTION
**Pull request by Imaraika**
- What does this PR do?
   - login via Google and Facebook

- Description of Task to be completed?

  - Setting up passport google Authentication configuration.
  - As a user I should be able to login via Google and Facebook
- How should this be manually tested?
  - Clone the repository.
  - In your home directory move into the cloned folder by cd space-bn-backend
  - Then run npm install to install all dependencies.
  - Run npm start to start the server.
- Any background context you want to provide?
  - N/A

- What are the relevant Pivotal Tracker stories?
   #175743317

- Screenshots (if appropriate)
   N/A
- Questions
  N/A

### CODE REVIEW

**Comment by Anguandia** [comment]
Please provide your environment variables on travis and write tests for your feature @Imaraika

**Comment by Anguandia** [comment]
Why do you have 2 different PRs for the same feature? Please delete the unnecessary extra

**Review by Anguandia — CHANGES_REQUESTED** [review]
Please incorporate tests

**Comment by Imaraika** [comment]
> Please incorporate tests
yes yes please ! I have worked on the tests! now all tests are passing

**Comment by Imaraika** [comment]
> Please incorporate tests
yes yes please ! I have worked on the tests! now all tests are passing

**Comment by Imaraika** [comment]
> Please provide your environment variables on travis and write tests for your feature @Imaraika

I have done this also ! thanks

**Comment by SamuelSabasNayo** [comment]
@Imaraika you can increase your test coverage, since your coverage of 94% is below the existing one, and it is required.

**Review by ConstantineMaliza — CHANGES_REQUESTED** [review]
great work @Imaraika  cover all test because it affects the test coverage on develop if it is not covered

**Review by SimonImanigirimpuhwe — COMMENTED** [review]
Resolve some conflicting files and then we are ready to go

File: locales/en.json [hunk 1]
```diff
@@ -58,6 +58,8 @@
 	"Request added successfully!": "Request added successfully!",
 	"Request updated successfully!": "Request updated successfully!",
 	"SUPER_ADMIN role connot be updated!": "SUPER_ADMIN role connot be updated!",
+	"Comment deleted successfully!": "Comment deleted successfully!",
+	"Request does not exist!": "Request does not exist!"
```
Comment (SamuelSabasNayo) [inline locales/en.json:62]: Put a comma at the end of this line.

File: src/routes/googleOuth.route.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+googleRouter.get(
+  "/google",
+  googleAuth.authenticate("google", { scope: ["profile", "email"] })
+);
+googleRouter.get(
+  "/google/callback",
+  googleAuth.authenticate("google"),
+  async (req, res) => {
+    const token = generateToken({
+      id: req.user.id,
```
Comment (SamuelSabasNayo) [inline src/routes/googleOuth.route.js:15]: You should use destructuring rather than repeating **req.user**.

**Review by SamuelSabasNayo — CHANGES_REQUESTED** [review]

**Comment by SamuelSabasNayo** [comment]
Please rebase.

**Pull request closed without merging**